File: ruby-impl/GameObject.rb
class GameObject
  attr_reader :name, :x, :y, :z, :rotation, :extra

  def initialize (name, x, y, z, rotation, extra)
    @name = name || ''
    @x = x || 0
    @y = y || 0
    @z = z || 0
    @rotation = rotation || 0
    @extra = extra || Array.new
  end
end

File: csharp-impl/HackathonLib/GameObject.cs
using System.Collections.Generic;

namespace HackathonLib
{
    public class GameObject
    {
        public double x, y, z;
        public double rotation;
        public string name;
        public List<object> extra;

        public GameObject(string name, double x, double y, double z, double rotation, List<object> extra)
        {
            this.name = name;
            this.x = x;
            this.y = y;
            this.z = z;
            this.rotation = rotation;
            this.extra = extra;
        }
    }
}
